airbyte-server/build.gradle

```diff
@@ -126,3 +129,36 @@ run {
     environment "AIRBYTE_ROLE", System.getenv('AIRBYTE_ROLE')
     environment "TEMPORAL_HOST", "localhost:7233"
 }
+
+task copyDocker(type: Copy) {
+    delete "build/docker"
+
+    from "${project.projectDir}/Dockerfile"
+    into "build/docker/"
+}
+
+task copyGeneratedTar(type: Copy) {
+    dependsOn copyDocker
+
+    from('build/distributions') {
+        include 'airbyte-server-*.tar'
+    }
+    into 'build/docker/bin'
+}
+
+def buildTag = System.getenv('VERSION') ?: 'dev'
+// Use task types
+task buildMyAppImage(type: DockerBuildImage) {
+    dependsOn copyDocker
+    dependsOn copyGeneratedTar
+    inputDir = file('build/docker')
+    images.add("airbyte/server:$buildTag")
+}
+assemble.dependsOn(buildMyAppImage)
+startScripts.dependsOn(shadowJar)
```

Comment:
These will all be cleaned up in the followup "make server build incemental" I hope?